test/dynamo/test_dicts.py

```diff
@@ -581,6 +581,37 @@ def forward(self, x):
         opt_mod = torch.compile(mod, backend="eager", fullgraph=True)
         self.assertEqual(mod(x), opt_mod(x))
 
+    def test_update_dunder_dict(self):
+        class UserDefined:
+            def run(self, x):
+                self.__dict__["a"] = 10
+                return x * self.a + self.__dict__["a"]
+
+        obj = UserDefined()
+
+        def fn(x):
+            return obj.run(x)
+
+        x = torch.randn(4)
+        ref = fn(x)
+        opt_fn = torch.compile(fn, backend="eager", fullgraph=True)
+        res = opt_fn(x)
+        self.assertEqual(ref, res)
```

Comment:
Ensure that you take `obj.a` has been updated after the compiled function is run (to check side effects).

Reply:
Updated.